pyccel/ast/cupyext.py

```diff
@@ -436,6 +444,38 @@ def __str__(self):
 
 #==============================================================================
 
+class CupyRavel(CupyArray):
+    """
+    Class representing a call to the cupy ravel function which
+    returns flattened version of the passed array
+
+    Parameters
+    ==========
+    arg : PyccelAstNode
+            A PyccelAstNode of unknown shape
+    memory_location : string
+            The location where the new array memory should be allocated
+    """
+    name   = 'ravel'
+    __slots__ = ()
+    def __new__(cls, arg):
+        if not isinstance(arg, (list, tuple, PyccelAstNode)):
+            raise TypeError('Unknown type of  %s.' % type(arg))
+        if arg.rank == 0:
+            raise TypeError('Unknown type of  %s.' % type(arg))
+        return super().__new__(cls)
+
+    def __init__(self, arg):
+        super().__init__(arg = arg)
+        shape = reduce((lambda x, y: x.python_value * y.python_value), self.shape)
+        self._shape = [shape if isinstance(shape, LiteralInteger) else LiteralInteger(shape)]
+        self._rank = 1
+
+    @property
+    def arg(self):
+        return self._arg
```

Comment:
How is this different from the function you are overriding?

Reply:
This one is my bad, i'll remove it.

Reply:
Removed in [def5e4e](https://github.com/pyccel/pyccel/pull/1222/commits/def5e4eb1e30a14de57c60d1033c70f4d0561c20).